### PR TITLE
New LogRecord protobuf BLE logging

### PR DESCRIPTION
**Pull request by thebentern**
Kissing cousins ticket for https://github.com/meshtastic/firmware/pull/4220

### CODE REVIEW

File: meshtastic/ble_interface.py [hunk 1]
```diff
@@ -82,6 +90,27 @@ def from_num_handler(self, _, b):  # pylint: disable=C0116
         self.should_read = True
 
     async def log_radio_handler(self, _, b):  # pylint: disable=C0116
+        log_record = mesh_pb2.LogRecord()
+        try:
+            log_record.ParseFromString(bytes(b))
+            log_record.message = log_record.message.replace("\n", "")
```
Comment (geeksville) [inline meshtastic/ble_interface.py:96]: What about instead of this we not include the new line on the sending side?  Looks great otherwise.

**Review by geeksville — APPROVED** [review]

**Comment by ianmcorvidae** [comment]
I think that eventually here we should do something like https://github.com/meshtastic/python/pull/607 is doing in that we should publish a pubsub message for log lines. That PR (WIP, of course) only passes along the log line and not the whole `LogRecord`, though.

I also have a question, which is whether any actual firmware releases went out with the now-legacy logging characteristic. We could possibly exclude it if there weren't any, but I haven't been watching that closely.

Other than my question and the suggestion from geeksville about removing newlines on the send side, I think this is good for now, and we can augment it with the pubsub once we've got something on both the BLE and the streaming interface sides.

**Comment by thebentern** [comment]
Sounds good. I'm moving the newline handling to the device

**Comment by geeksville** [comment]
I agree about pubsubish cleanup of the Log system on the device side.
Because it really never should have been in RediectablePrint and has grown
quite a bit.  I especially want to fix the thing Andre mentioned about
logging also working over the TCP endpoint.

I have a PR half done for this.  I was hoping to send it in before I left
town for a two day camping trip.  I'll be back tonight and try to send it
ASAP.

Also yeah: if we haven't done an official release with the initial logging
over BLE feature hopefully we only need to support both variants briefly.
Because it costs an extra 512 byte buffer at the BLE level.

(Sent from a phone - please ignore typos)

**Pull request merged**